Content.Server/Salvage/SalvageSystem.cs

```diff
@@ -0,0 +1,310 @@
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.Threading;
+using Content.Server.Administration.Logs;
+using Content.Server.Chat.Managers;
+using Content.Server.GameTicking;
+using Content.Shared.Administration.Logs;
+using Content.Shared.CCVar;
+using Content.Shared.Database;
+using Content.Shared.Examine;
+using Content.Shared.GameTicking;
+using Content.Shared.Interaction;
+using Content.Shared.Hands.Components;
+using Content.Shared.Popups;
+using Robust.Server.Player;
+using Robust.Server.Maps;
+using Robust.Shared.Audio;
+using Robust.Shared.Configuration;
+using Robust.Shared.Map;
+using Robust.Shared.Log;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+using Robust.Shared.GameObjects;
+using Robust.Shared.ViewVariables;
+using Robust.Shared.IoC;
+using Robust.Shared.Maths;
+using Robust.Shared.Localization;
+using Robust.Shared.Player;
+using Robust.Shared.Physics;
+using Robust.Shared.Timing;
+
+namespace Content.Server.Salvage
+{
+    public class SalvageSystem : EntitySystem
+    {
+        [Dependency] private readonly IChatManager _chatManager = default!;
+        [Dependency] private readonly IMapLoader _mapLoader = default!;
+        [Dependency] private readonly IMapManager _mapManager = default!;
+        [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+        [Dependency] private readonly IConfigurationManager _configurationManager = default!;
+        [Dependency] private readonly SharedPhysicsSystem _physicsSystem = default!;
+        [Dependency] private readonly IRobustRandom _random = default!;
+        [Dependency] private readonly SharedPopupSystem _popupSystem = default!;
+
+        [ViewVariables]
+        public EntityUid PulledObject = EntityUid.Invalid;
+
+        [ViewVariables]
+        public SalvageSystemState State = SalvageSystemState.Inactive;
+
+        [ViewVariables]
+        public float StateTimer = 0.0f;
+
+        public const float PullInTimer = 2.0f;
+        public const float HoldTimer = 240.0f;
+        public const float LeaveTimer = 60.0f;
+        public const float AngularVelocityRangeRadians = 0.25f;
```

Comment:
Realistically it should be spawning further out and getting pulled in for longer. Also AngularVelocityRangeRadians isn't used anymore after I said it just shouldn't be spinning constantly.